html5lib/_inputstream.py

```diff
@@ -668,15 +668,11 @@ def matchBytes(self, bytes):
     def jumpTo(self, bytes):
         """Look for the next sequence of bytes matching a given sequence. If
         a match is found advance the position to the last byte of the match"""
-        newPosition = self[self.position:].find(bytes)
-        if newPosition > -1:
-            # XXX: This is ugly, but I can't see a nicer way to fix this.
-            if self._position == -1:
-                self._position = 0
-            self._position += (newPosition + len(bytes) - 1)
-            return True
-        else:
+        try:
+            self._position = self.index(bytes, self.position) + len(bytes) - 1
```

Comment:
Did you test the tradeoff between `.find()` here and `.index()`? The former could avoid constructing an exception in the no-match case, but will need some additional logic.

Reply:
`index` is a minor win; it's pretty unusual there's no match, and it avoids the branch in the interpreted code in every case.